Feat/dv-12/registration-page

marynakernychna opened the pull request.

marynakernychna committed:
Add registration page layout

> Add a path to the registration page
>Add the registration page layout
> Create common styles for login and registration pages and separately for all other pages
> Add previously removed React.StrictMode
> Align code in previously added files

marynakernychna committed:
Edit registration page layout

> Add an adaptation to the registration page
> Add links to the Login and Home pages
> Add authentication service
> Add a general style file

YevheniiVovk commented on public/index.html:
```diff
@@ -1,15 +1,16 @@
+  <meta charset="utf-8" />
+  <meta name="description" content="Web site created using create-react-app" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>React App</title>
```
Why it still here. Do you create the  react application?

marynakernychna replied:
I formatted the code and Git marked their as new lines

YevheniiVovk commented on public/index.html:
```diff
@@ -1,15 +1,16 @@
+  <div id="root"></div>
+</body>
+
+</html>
```
Empty line at EoF

YevheniiVovk commented on public/index.html:
```diff
@@ -1,15 +1,16 @@
+</head>
+
+<body>
+  <noscript>You need to enable JavaScript to run this app.</noscript>
```
Let's give more context for user. Like you need to use JS to run logistman app or ect. Don't use template verbs.

YevheniiVovk commented on src/components/authentication/styles.module.css:
```diff
@@ -0,0 +1,204 @@
+    }
+}
+
+/*( > 1200  ) ( 1199 - 600 ) ( 599 < ) - sizes of screen*/ 
```
Empty line at EoF

YevheniiVovk commented on src/services/alert.service.js:
```diff
@@ -1,59 +1,59 @@
   }
 
-  static successMessage(title = 'Success'){
+  static successMessage(title = 'Success') {
```
Possibly we should move timer value like a title argument

YevheniiVovk commented on src/styles.css:
```diff
@@ -0,0 +1,4 @@
+body {
+    height: 100vh;
+    background-color: #093545;
+}
```
Empty line at End of File

marynakernychna committed:
Edit adaptability, styles and correction of comments

> expand <noscript>'s text
> add empty lines at the end of files
> edit adaptability
> delete general style.css file
> move timer value in success message to the arguments

marynakernychna committed:
Add validation, edit adaptability and fix service

> Connect pre-forgotten Ant Design styles
> Name the registration page
> Add form validations
> Edit adaptability
> Add pre-forgotten field to the success message alert

vladtymo commented on src/App.js:
```diff
@@ -1,11 +1,17 @@
+          <Registration />
+        </Route>
+
         <Route path="/" >
```
Why don't you put the exact attribute to the root path?

vladtymo commented on src/components/authentication/registration/index.jsx:
```diff
@@ -0,0 +1,158 @@
+                                    return Promise.reject(
+                                        new Error
+                                            (
+                                                'The two passwords that you entered do not match!'
```
Consider moving the error messages to a separate file.

marynakernychna committed:
Add work with server endpoint

> Change project title
> Create authentication service
> Add history
> Create error, input and url constants
> Bind with server endpoint

VadimChorrny commented on src/components/authentication/styles.module.css:
```diff
@@ -0,0 +1,191 @@
+    }
+
+    .submitItem div {
+        height: 34px !important;
```
what is this? !important?

vadymchornyidev replied:
perhaps !important need change or delete

marynakernychna commented on src/components/authentication/styles.module.css:
```diff
@@ -0,0 +1,191 @@
+    }
+
+    .submitItem div {
+        height: 34px !important;
```
since this is not critical, it can be fixed later

YevheniiVovk commented on src/services/alert.service.js:
```diff
@@ -1,59 +1,59 @@
+      confirmButtonText: confirmButtonText,
+      denyButtonText: denyButtonText,
+    }).then((result) => {
+      if (result.isConfirmed) {
+        this.successMessage(confirmedText)
+        return false
+      } else if (result.isDenied) {
+        Swal.fire(DenyText, '', 'info')
+        return true
```
Why we write if to hardcode return value

YevheniiVovk commented on src/services/alert.service.js:
```diff
@@ -1,59 +1,59 @@
+    })
   }
 
   static confirmMessage(title = 'Are you sure?', text = 'Are you realy sure?',
```
are we sure that this should be static ?

pashunskyi replied:
without static methods in other classes are not available

YevheniiVovk replied:
It's clear. You should inject the instance if you remove the static. 
The answer for this question "are we sure that this should be static ?" should be based on opinion with facts that describe the technical solution and provide the best practice in scope of creating wrapper methods for Swal with preset configuration.

YevheniiVovk commented on src/constants/inputRegexes.js:
```diff
@@ -0,0 +1,5 @@
+export const inputRegexes = Object.freeze({
+    NAME: "^[A-Z][a-z]{1,51}$",
+    SURNAME: "^[A-Z][a-z]{1,51}$",
+    PASSWORD: "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*?&])[A-Za-z0-9@$!%*?&]{7,51}$"
```
let's find the way to set regex for each model

YevheniiVovk commented on src/components/authentication/styles.module.css:
```diff
@@ -0,0 +1,191 @@
+}
+
+.passwordInput {
+    background-color: #224957 !important;
+    border-radius: 10px !important;
+    border: none !important;
+    outline-width: 0;
+    color: white;
+    height: 40px;
+    font-size: 18px;
+    width: 280px !important;
+    padding: 0px 10px;
+}
+
+.textForm ::placeholder,
+.passwordForm ::placeholder {
+    color: white;
+    opacity: .7;
+    font-size: medium;
+}
+
+.submitItem div {
+    height: 40px !important;
+}
+
+.submitItem {
+    margin-bottom: 10px !important;
+    margin-top: 35px !important;
+}
+
+.submitButton {
+    background-color: #17a15c !important;
+    border: none !important;
+    height: 40px !important;
+    width: 280px;
+    font-size: 16px !important;
```
rewrite styles for remove !important and never use it.
